python/rascaline/transformer.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+from equistore import Labels, TensorBlock, TensorMap
+
+
+class Transformer:
+    """The 'Transformer' class makes it easy to create a representation matrix
+    when using some other matrix as a reference. A classic use case is to create
+    a TensorMap representation for a dataset, then perform transformations
+    within that TensorMap (e.g., keys_to_features or keys_to_properties), and
+    select the most useful features in the transformed TensorMap.
+    The 'Transformer' allows a set of these features to be used to calculate
+    a new TensorMap, thus saving computation time and maintaining a single
+    representation for all representations.
+
+    Parameters:
+    -----------
+    :param selectors: The selector to be used when selecting the features.
+        Currently the same selector is used for different blocks.
+        #TODO: It should be possible for the user to pass a list of selectors
+        for blocks.
+    :param transformation: the type of transformation to be performed.
+        Two options are possible - 'keys_to_features' and 'keys_to_properties'.
+        #TODO: provide the ability to pass a list of conversions that will occur
+        one after the other.
+    :param moved_keys: Those keys which will be moved during the transformation.
+        This variable can accept knowledge of type str (one key), list (a list
+        of keys) and Labels (in addition to the name of the keys, pass a list
+        of keys to be moved).
+    """
+
+    def __init__(self, selector, transformation=None, moved_keys=None):
+        #
+        self.selector = selector
+        self.transformation = transformation
+        self.moved_keys = moved_keys
```

Comment:
The user should not access these after creating an instance of this class

```suggestion
        self._selector = selector
        self._transformation = transformation
        self._moved_keys = moved_keys
```